### PR TITLE
feat(aws): Allow self-referencing security group clone ops

### PR DESCRIPTION
**Pull request by robzienert**
Performs an upsert in a two-phase operation if the security group has self-referencing rules and does not already exist.

### CODE REVIEW

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/securitygroup/SecurityGroupIngressConverter.groovy [hunk 1]
```diff
@@ -75,7 +85,10 @@ class SecurityGroupIngressConverter {
         ipPermissions.add(newIpPermission)
       }
     }
-    new ConvertedIngress(ipPermissions, missing)
+    new ConvertedIngress(ipPermissions, new MissingSecurityGroups(
+      all: missing,
+      selfReferencing: missing.findAll { it.name == description.name }
```
Comment (robzienert) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/securitygroup/SecurityGroupIngressConverter.groovy]: Should maybe probably add more equality checks here?

Comment (ajordens) [reply clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/securitygroup/SecurityGroupIngressConverter.groovy]: account/userId maybe?

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/securitygroup/UpsertSecurityGroupAtomicOperation.groovy [hunk 2]
```diff
@@ -112,4 +109,19 @@ class UpsertSecurityGroupAtomicOperation implements AtomicOperation<Void> {
     null
   }
 
+  private ConvertedIngress convertDescriptionToIngress(SecurityGroupLookup securityGroupLookup, UpsertSecurityGroupDescription description, boolean failOnMissingSelfReferencingRules) {
+    ConvertedIngress ipPermissionsFromDescription = SecurityGroupIngressConverter.
+      convertIngressToIpPermissions(securityGroupLookup, description)
+
+    if (ipPermissionsFromDescription.missingSecurityGroups.hasMissingNonSelfReferencingGroups() && !failOnMissingSelfReferencingRules) {
```
Comment (ajordens) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/securitygroup/UpsertSecurityGroupAtomicOperation.groovy]: Is this logic correct ... throw an exception if we have missing non-self referential groups and `failOnMissingSelfReferential == false`?

Comment (robzienert) [reply clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/securitygroup/UpsertSecurityGroupAtomicOperation.groovy]: Yeah, not sure where my head was at there. Updated.

**Pull request merged**